.github/workflows/development-environment.yml

```diff
@@ -46,18 +49,18 @@ jobs:
         continue-on-error: true
         # Xcode CLI & brew are already installed, thus, no need to call xcode-select install
         # XXX: Can the Brew set up be cached?
+        # If you want to speed up the CI for your PR, you can use --no-upgrade and HOMEBREW_NO_AUTO_UPDATE=1
         run: |
           brew bundle
 
       - name: Install Python (via Pyenv)
         id: python-version
-        shell: bash
-        # XXX: Locally, pyenv makes 3.6.10 be the selected version in use,
-        # however, in here it selects /usr/local/bin/python. I'm using `pyenv global` to work around it
         # XXX: Can we cache the installed Python environments?
+        # We echo to $GITHUB_ENV to make the PATH changes by pyenv init permanent for the rest of the execution
         run: |
           make setup-pyenv
-          pyenv global 3.6.10
+          eval "$(pyenv init -)"
+          echo "PATH=$PATH" >> $GITHUB_ENV
```

Comment:
This intends to make the PATH changes permanent for the rest of the workflow, thus, not needing to call `source` on every step we want `pyenv` to load 3.6.10